enhancements/running-on-the-host.md

```diff
@@ -0,0 +1,167 @@
+
+- running the router pod, in charge of providing the network control plane with FRR and the data plane with veths and the underlay interface
+- running the controller pod, in charge of reading the static configuration and receiving the additional configuration when the kubernetes api is available
+
+#### Handling the lifecycle of the components
+
+The router and the controller need to run as systemd units. This allows us to run them before the kubelet and the CNI start.
+Once the machinery is implemented, alternative ways to run them can be added for those environments where systemd is not enabled.
+
+#### Running on the host
+
+Podman provides a way to maintain the current pod structure while running the pods as systemd units. This allow us to keep reasoning of pods and
+to have a good parallel between the "Kubernetes mode" and the "Host mode".
+
+Because of this, the initial implementation will be based on Podman. As per systemd above, when we have a way of running the pods on the host, it should be
+relatively easy to adapt to other container runtimes.
+
+#### Interactions
+
+There are some interactions between the controller process and the router (and the surrounding system) that need to be adapted to the new scenario:
+
+##### Retrieving the router's network namespace
+
+We are using the container runtime api to retrieve the network namespace of the pod.
+When running on the host, the target namespace can be reitrieved by telling podman to fill a pid file, and by using the pid to retrieve the network namespace of the process.
+
+##### Sending a signal to the reloader container to reload the FRR configuration
+
+Currently the controller retrieves the target pod's IP using the kubernetes API. When running on the host, a unix socket can be shared between the containers.
+
+##### Restarting the router pod
+
+The controller pod currently deletes the router pod to restart the underlay creation machinery. When running on the host, the controller pod can interact either with Podman or with Systemd to restart the pod.
+
+#### Consuming the static configuration
+
+The static configuration must be provided to the OpenPERouter as files in a known location. The configuration is composed by:
+
+- A static configuration with the same structure of the CRDs (ie underlays, vnis) to be used for bringing up the basic connectivity
+- An extra configuration to fill the requirements of running on the host. At the time of writing, the only extra configuration required seem to be a node index, to be associated manually (or by automation) to be different per each node.
+
+#### Consuming the Kubernetes API
+
+Running the controller on the host while it's still be able to access the kubernetes API is a challenge. One possible way to solve this is to have a proxy pod running on the cluster and translating its credentials to the controller running on the host.
```

Comment:
Similar situation happend when CNI's need to access api server, what they have is the kubeconfig stored under their static config dir /etc/cni.d.

We can do similar here, when kubernetes start a openperouter pod just generate the kubeconfig and store it under the openperouter static config the the controller can use it, this way we don't need a running proxy pod at kubernetes it will be a one shot installer pod.